src/UI/Implementation/Component/Input/Input.php

```diff
@@ -310,6 +359,24 @@ private function applyOperationsTo($res) {
 	}
 
 	/**
+	 * Get the operations that should be performed on the input.
+	 *
+	 * @return (Transformation|Constraint)[]
+	 */
+	private function getOperations() {
+		if ($this->isRequired()) {
+			$op = $this->getConstraintForRequirement();
+			if ($op !== null) {
+				yield $op;
```

Comment:
Does this work in PHP5? Should you not write (yield $op); for backwards compatibility?

Reply:
This works with 5.6.

Reply:
oke

Reply:
would this still work if getConstraintForRequirement also used yield. I have in mind there was something of an indexing issue with yield if used kind of recursively.

Reply:
If `getConstraintForRequirement` would `yield`, `$op` wouldn't be a value but a generator. You would need to generate values with the generator (with `foreach` or by other means) and `yield` them as well. I think PHP 7 has a `yield from` shortcut for that.